fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -1415,14 +1415,22 @@ private CommandProcessingResult updateDatatableEntry(final String dataTableName,
             }
         }
 
-        return new CommandProcessingResultBuilder() //
+        CommandProcessingResultBuilder result = new CommandProcessingResultBuilder() //
+                .withCommandId(command.commandId()) //
                 .withOfficeId(commandProcessingResult.getOfficeId()) //
                 .withGroupId(commandProcessingResult.getGroupId()) //
                 .withClientId(commandProcessingResult.getClientId()) //
                 .withSavingsId(commandProcessingResult.getSavingsId()) //
                 .withLoanId(commandProcessingResult.getLoanId()) //
-                .with(changes) //
-                .build();
+                .with(changes);
+
+        if (command.subentityId() != null) {
+            result = result.withEntityId(command.subentityId());
```

Comment:
Why don't we simply set the subResourceId in this case?

Reply:
@galovics what you mean? Currently there is not any method in the `JsonCommand` class to return the `subresourceId` attribute, I mean the `subentityId()` method returns that attribute 

<img width="401" alt="Screenshot 2022-11-21 at 13 25 02" src="https://user-images.githubusercontent.com/44206706/203141768-75c48196-9a54-4447-98ab-df22c0fe0e22.png">

Reply:
I think we dont need these conditions.
`
CommandProcessingResultBuilder result = new CommandProcessingResultBuilder()
.withEntityId(command.entityId()
.withSubEntityId(command.subentityId()
...
`

Reply:
subEntity == subResourceId

Reply:
Personally i would do this:
```
new CommandProcessingResultBuilder().withEntityId(datatableId != null? datatableId : appTableId);
```

Reply:
Done